### DISCUSSION TITLE
Implicit constructor in C#

### DISCUSSION THREAD

**Discussion by paroj** [swig/swig#1927] [General]
I use the following typemap for Python
```cpp
%typemap(in) const Ogre::Radian& (float tmp) {
    void *argp = 0;
    int res = SWIG_ConvertPtr($input, &argp, $descriptor, $disown);
    if (SWIG_IsOK(res)) {
        $1 = ($ltype)(argp);
    } 
    else {
        res = SWIG_AsVal_float($input, &tmp);
        $1 = (Ogre::Radian*)&tmp; // dont judge

        if (!SWIG_IsOK(res))
            SWIG_exception_fail(SWIG_TypeError, "Expected float or Ogre::Radian");
    }
}
```

but for C#, I get undefined errors for `SWIG_IsOK`, `SWIGTYPE_p_Ogre__Radian` etc.

I get the feeling this is not the best way for C#, but I would expect it to be compatible at least.

I also tried using implicit.swg, but that gave me syntax errors in `%implicit(Ogre::Radian, float);`

**Comment by wsfulton** [comment]
Typemaps are language specific, so you need a C# specific typemap. However, note that many of the scripting languages implement a set of SWIG functions such as `SWIG_ConvertPtr` so can sometimes be written to support multiple languages.
